presto-main/src/main/java/com/facebook/presto/execution/resourceGroups/InternalResourceGroup.java

```diff
@@ -682,6 +690,7 @@ public void run(ManagedQueryExecution query)
                 canRun &= group.canRunMore();
                 if (!canQueue && !canRun) {
                     query.setResourceGroupQueuedOn(group.getId());
+                    query.setQueuingReason(Optional.of(QUEUE_FULL));
```

Comment:
i don't think we need to set this for a failed queries, can't we just keep resourceGroupQueuedOn optional and the reason also empty?